File: learning-algorithm/src/main/java/io/vincent/learning/stack/algorithm/leetcode/ListNode.java
package io.vincent.learning.stack.algorithm.leetcode;

/**
 * leetcode ListNode class.
 *
 * @author Vincent
 * @since 1.0, 2019/3/24
 */
public class ListNode {
    int value;
    ListNode next;

    ListNode(int value) {
        this.value = value;
        this.next = null;
    }

    public ListNode(int value, ListNode next) {
        this.value = value;
        this.next = next;
    }

    public String toString() {
        return "ListNode{" +
                       "value=" + value +
                       ", next=" + next +
                       '}';
    }

    @Override
    public int hashCode() {
        return super.hashCode();
    }
}

File: learning-algorithm/src/main/kotlin/io/vincent/learning/stack/algorithm/leetcode/ListNode.kt
package io.vincent.learning.stack.algorithm.leetcode

/**
 * Created by Vincent on 2019/3/24.
 * @author Vincent
 * @since 1.0, 2019/3/24
 */
class ListNode(var value: Int, var next: ListNode? = null) {
    override fun toString(): String {
        return "ListNode{" +
                "value=" + value +
                ", next=" + next +
                '}'.toString()
    }
}
